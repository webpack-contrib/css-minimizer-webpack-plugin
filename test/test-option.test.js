import CssMinimizerPlugin from "../src/index";

import {
  getCompiler,
  compile,
  readAssets,
  getErrors,
  getWarnings,
} from "./helpers";

describe('when applied with "test" option', () => {
  let compiler;

  beforeEach(() => {
    compiler = getCompiler({
      entry: {
        bar1: `${__dirname}/fixtures/test/bar1.css`,
        bar2: `${__dirname}/fixtures/test/bar2.css`,
        foo: `${__dirname}/fixtures/test/foo.css`,
      },
    });
  });

  it("matches snapshot with empty value", async () => {
    new CssMinimizerPlugin().apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it('matches snapshot for a single "test" value (RegExp)', async () => {
    new CssMinimizerPlugin({
      test: /bar.*\.css$/,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it('matches snapshot for multiple "test" value (RegExp)', async () => {
    new CssMinimizerPlugin({
      test: [/bar1.*\.css$/, /bar2.*\.css$/],
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });
});
