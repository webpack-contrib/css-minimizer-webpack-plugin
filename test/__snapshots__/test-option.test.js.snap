// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with "test" option matches snapshot for a single "test" value (RegExp): assets 1`] = `
{
  "bar1.css": "body{color:red;font-size:20px}a{color:blue}",
  "bar2.css": "h1{color:green}h2{color:#ff0}",
  "foo.css": "a {
  text-align: center;
}

",
}
`;

exports[`when applied with "test" option matches snapshot for a single "test" value (RegExp): errors 1`] = `[]`;

exports[`when applied with "test" option matches snapshot for a single "test" value (RegExp): warnings 1`] = `[]`;

exports[`when applied with "test" option matches snapshot for multiple "test" value (RegExp): assets 1`] = `
{
  "bar1.css": "body{color:red;font-size:20px}a{color:blue}",
  "bar2.css": "h1{color:green}h2{color:#ff0}",
  "foo.css": "a {
  text-align: center;
}

",
}
`;

exports[`when applied with "test" option matches snapshot for multiple "test" value (RegExp): errors 1`] = `[]`;

exports[`when applied with "test" option matches snapshot for multiple "test" value (RegExp): warnings 1`] = `[]`;

exports[`when applied with "test" option matches snapshot with empty value: assets 1`] = `
{
  "bar1.css": "body{color:red;font-size:20px}a{color:blue}",
  "bar2.css": "h1{color:green}h2{color:#ff0}",
  "foo.css": "a{text-align:center}",
}
`;

exports[`when applied with "test" option matches snapshot with empty value: errors 1`] = `[]`;

exports[`when applied with "test" option matches snapshot with empty value: warnings 1`] = `[]`;
